From 4da01cb96d838fe805a71d27eb07b41bbd4d49e8 Mon Sep 17 00:00:00 2001
From: Oliver Eftevaag <oliver.eftevaag@qt.io>
Date: Fri, 19 Sep 2025 17:31:01 +0200
Subject: [PATCH 509/553] Windows11: Don't force highlightedText to be
 windowText in high contrast

The style would overwrite the highlightedText color to be equal to the
windowText for the active group. This would negatively impact styles in
Qt Quick, whenever a QApplication was being used. The highlightedText
color would not be correct.

Task-number: QTBUG-140507
Change-Id: I282971a4556d0b10d9e32af3514aa6f51800e35e
Reviewed-by: Richard Moe Gustavsen <richard.gustavsen@qt.io>
Reviewed-by: Wladimir Leuschner <wladimir.leuschner@qt.io>
(cherry picked from commit 0c337016be87a7370f15c95f51421a873e3b97e2)
---
 src/plugins/styles/modernwindows/qwindows11style.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/plugins/styles/modernwindows/qwindows11style.cpp b/src/plugins/styles/modernwindows/qwindows11style.cpp
index b6d0dd3c426..dd5867b6914 100644
--- a/src/plugins/styles/modernwindows/qwindows11style.cpp
+++ b/src/plugins/styles/modernwindows/qwindows11style.cpp
@@ -2425,9 +2425,6 @@ void QWindows11Style::polish(QPalette& result)
     SET_IF_UNRESOLVED(QPalette::Inactive, QPalette::HighlightedText, result.highlightedText().color());
     SET_IF_UNRESOLVED(QPalette::Inactive, QPalette::Text, result.text().color());
     SET_IF_UNRESOLVED(QPalette::Inactive, QPalette::WindowText, result.windowText().color());
-
-    if (highContrastTheme)
-        result.setColor(QPalette::Active, QPalette::HighlightedText, result.windowText().color());
 }
 
 QColor QWindows11Style::calculateAccentColor(const QStyleOption *option) const
-- 
2.51.2

